Formula/pyside.rb

```diff
@@ -58,7 +59,11 @@ def install
               "in_build = Path(\"@CMAKE_BINARY_DIR@\") in location.parents",
               "in_build = Path(\"@CMAKE_BINARY_DIR@\").resolve() in location.parents"
 
+    # Install python scripts into pkgshare rather than bin
+    inreplace "sources/pyside-tools/CMakeLists.txt", "DESTINATION bin", "DESTINATION #{pkgshare}"
+
     args = std_cmake_args + [
+      "-DCMAKE_CXX_COMPILER=#{ENV.cxx}",
```

Comment:
This might fix `HOMEBREW_LIBRARY` reference or can inreplace (`pyside@2` does both for some reason).

I think C++ compiler path is getting saved by `sources/shiboken6/ApiExtractor/clangparser/compilersupport.cpp`:
```cxx
#ifdef CMAKE_CXX_COMPILER
    if (platform() != Platform::macOS)
        result = QString::fromLocal8Bit(CMAKE_CXX_COMPILER);
#endif
```
`sources/shiboken6/ApiExtractor/CMakeLists.txt`:
```cmake
target_compile_definitions(apiextractor
                           PRIVATE CMAKE_CXX_COMPILER="${CMAKE_CXX_COMPILER}"
                           PRIVATE QT_LEAN_HEADERS=1)
```

Reply:
This seems to bypass the shim, which is not what we want:

> Check for working CXX compiler: /usr/local/opt/llvm/bin/clang++ - skipped

Reply:
Why is it picking up that `clang++`?

This is one of the workarounds we added to `pyside@2`. We could try a variation of other workaround from that formula (maybe along with PR to migrate to `python@3.11` if it is compatible):
```rb
      # Avoid shim reference.
      inreplace "sources/shiboken2/ApiExtractor/CMakeLists.txt",
                "${CMAKE_CXX_COMPILER}",
                "/usr/bin/c++"
```

Reply:
I'm guessing CMake doesn't read the PATH like expected. Basically avoid `-DCMAKE_C(XX)_COMPILER`.

That shim reference fix makes sense to me.

> maybe along with PR to migrate to python@3.11 if it is compatible

Due to the use of Limited ABI, we should technically be able to build it once (for the oldest version?) for all Python but there may well be some complications given shiboken exposes way too much internals - there may well be Python version checks in the installed headers that will need checking.